crates/transaction-generator-lib/src/lib.rs

```diff
@@ -90,6 +95,231 @@ pub trait TransactionGenerator: Sync + Send {
     ) -> Vec<SignedTransaction>;
 }
 
+pub trait TxnPatternGenerator: Send + Sync {
+    /*
+        1. The class implementing the trait should ensure that the pool is large enough.
+        2. The trait is independent of the contents of the pool. In case the contents of the pool
+           matter, then the class implementing the trait should handle it.
+        3. The trait returns: array of [senders, [multi_sig_signers], [receivers]]
+    */
+    fn generate_tx_pattern(
+        &mut self,
+        rng: &mut StdRng,
+        pool_start: usize,
+        pool_size: usize,
```

Comment:
these seems like arguments for the constructor of the below implementations, not something all implementations care about?